File: BinDec/b2d_d2b.java
import java.util.*;

class HelloWorld
{
	static int decimal(String bin)
	{
		char[] i_bits = bin.toCharArray();
		int sz = i_bits.length, dec=0, index;
		index = sz - 1;

		for (char c : i_bits)
		{
			if (c == '1')
			{
				dec += Math.pow(2, index);
			}

			index--;
		}

		return dec;
	}

	static String binary(int dec)
	{
		List<String> bits = new ArrayList<String>();
		String bin = "";

		while (dec > 0)
		{
			bits.add(""+(dec % 2));
			dec /= 2;
		}

		for (int i = bits.size()-1; i >= 0; i--)
		{
			bin += bits.get(i);
		}

		return bin;
	}

	public static void main(String[] args)
	{
		Scanner sc = new Scanner(System.in);

		char mode;

		System.out.print("Enter input type(b/d): ");
		mode = sc.nextLine().charAt(0);

		if (mode == 'b')
		{
			String bin;

			System.out.print("Enter binary value: ");
			bin = sc.nextLine();

			System.out.println("Decimal: " + decimal(bin));
		}

		else
		{
			int dec;

			System.out.print("Enter a decimal value: ");
			dec = sc.nextInt();

			System.out.println("Binary: " + binary(dec));
		}

		sc.close();
	}
}

File: BinDec/b2d_d2b.cs
using System;

namespace SecreteMissionCodes
{
    class MainClass
    {
        static int Decimal(string bin)
        {
            int sz, dec=0, index;

            sz = bin.Length;
            index = sz - 1;

            foreach (char bit in bin)
            {
                if (bit == '1')
                {
                    dec += Convert.ToInt32(Math.Pow(2, index));
                }

                index--;
            }

            return dec;
        }

        static string Binary(int dec)
        {
            string bits = "", bin = "";

            while (dec > 0)
            {
                bits += $"{dec % 2}";
                dec /= 2;
            }

            for (int i = bits.Length-1; i >= 0; i--)
            {
                bin += bits[i];
            }

            return bin;
        }

        public static void Main(string[] args)
        {
            char mode;

            Console.Write("Enter input type: ");
            mode = Console.ReadLine()[0];

            if (mode == 'b')
            {
                string bin;

                Console.Write("Enter binary value: ");
                bin = Console.ReadLine();

                Console.WriteLine($"Decimal: {Decimal(bin)}");
            }

            else
            {
                int dec;

                Console.Write("Enter the decimal input: ");
                dec = Convert.ToInt32(Console.ReadLine());

                Console.WriteLine($"Binary: {Binary(dec)}");
            }
        }
    }
}
